### PR TITLE
Add github workflow for building autogenerated documentation

### PR DESCRIPTION
**Pull request by adellibovi**
### Short description 📝

Hello, this PR introduces, under beta, a new workflow to generate the documentation of tuist manifests API, automatically generated from source code using DoCC.

The documentation is currently uploaded under
* https://tuist.github.io/tuist/main/documentation/projectdescription/
* https://tuist.github.io/tuist/3.0.1/documentation/projectdescription/
* https://tuist.github.io/tuist/3.0.0/documentation/projectdescription/
* etc...

I would merge this as is, and we can start improving how the documentation looks.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [x] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: .github/workflows/docc.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                mkdir -p .build/symbol-graphs;
+                swift build \
+                  --target ProjectDescription \
+                  -Xswiftc \
+                  -emit-symbol-graph \
+                  -Xswiftc \
+                  -emit-symbol-graph-dir \
+                  -Xswiftc \
+                  .build/symbol-graphs \
+                  && swift-docc/.build/release/docc convert Sources/ProjectDescription/Documentation.docc \
```
Comment (danyf90) [inline .github/workflows/docc.yml:87]: Why do we need the `&&`?

File: .github/workflows/docc.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                    --fallback-display-name ProjectDescription \
+                    --fallback-bundle-identifier io.tuist.ProjectDescription \
+                    --fallback-bundle-version 0.0.0 \
+                    --additional-symbol-graph-dir \
+                    .build/symbol-graphs \
+                    --transform-for-static-hosting \
+                    --hosting-base-path /tuist/"$tag" \
+                    --output-path docs-out/"$tag" \
+                    && echo "✅ Documentation generated for "$tag" release." \
+                    || echo "⚠️ Documentation skipped for "$tag".";
```
Comment (danyf90) [inline .github/workflows/docc.yml:97]: Not sure I got the logic behind `&&`  and `||` here 🤔

**Review by danyf90 — APPROVED** [review]

File: .github/workflows/docc.yml (hunk 2 above)
Comment (adellibovi) [reply .github/workflows/docc.yml:97]: It's used for success/failure

**Review by kwridan — APPROVED** [review]
Nice work @adellibovi !

Only small suggestion I would add here is perhaps extracting some of the commands / parts to shell scripts (or maybe Fourier) which could potentially allow us to easily produce some of the steps locally (e.g. just the local rendering of the different tags)

Just a thought 💭

**Comment by adellibovi** [comment]
> This looks a lot more complicated than it needs to be.
> 
> Why run on ubuntu and download the whole toolchain when we can run on macOS and just use `xcodebuild docbuild`?

Unfortunately, there are some parameters required for the static hosting not available in the current Xcode version, we would need to build anyway DocC, so at this point, we can have a Linux image that is generally faster and have a different parallel queue.

**Comment by adellibovi** [comment]
> Nice work @adellibovi !
> 
> Only small suggestion I would add here is perhaps extracting some of the commands / parts to shell scripts (or maybe Fourier) which could potentially allow us to easily produce some of the steps locally (e.g. just the local rendering of the different tags)
> 
> Just a thought 💭

Thanks Kas! I would maybe add it to contributors' documentation (when we officially move to the new documentation) that it can be built directly from Xcode ("Build documentation" command), that would be the preferred way and easistes to try out changes.

**Pull request merged**